test.js

```diff
@@ -146,6 +146,14 @@ test('Function.prototype.apply with the arguments object', t => {
   t.true(foo('[string|number] <object>', 'bar', {}));
 });
 
+test('allows `this` to be the typeConfig string', t => {
+  function foo () {
+    return argsert.apply('<null|object> [number]', arguments);
```

Comment:
`apply` takes arguments in the form `this, [argsArray]`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply

so what I'm doing here is somewhat slight of hand by setting `this = '<null|object> [number]'`.